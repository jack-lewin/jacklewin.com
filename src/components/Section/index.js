import React from 'react';
import { HTMLContent } from '../Content';
import SectionProps from '../../prop-types/components/section';
import styles from './styles.module.scss';

const Section = ({ level, title, content }) => {
  const Heading = `h${level}`;
  const headingClass = level === 1 ? styles.heading : styles.subheading;

  return (
    <section className={styles.container}>
      <Heading className={headingClass}>{title}</Heading>
      {typeof content === 'string' ? <HTMLContent content={content} /> : content}
    </section>
  );
};

Section.propTypes = SectionProps;

export default Section;
