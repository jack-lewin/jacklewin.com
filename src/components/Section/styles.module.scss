@import '../../functions.scss';

.container {
  flex-grow: 1;
  max-width: 800px;
  text-align: justify;
}

.heading {
  margin: spacing(large) 0;
  font-size: 1.8em;
}

.subheading {
  margin: spacing(large) 0 0;
  font-size: 1.2em;
}
