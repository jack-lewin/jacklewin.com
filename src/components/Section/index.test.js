import React from 'react';
import { shallow } from 'enzyme';
import Section from '.';

describe('Section', () => {
  const baseProps = {
    level: 3,
    title: 'Title goes here',
    content: '<p>my dangerous html</p>',
  };

  it('should render as expected', () => {
    const element = shallow(<Section {...baseProps} />);
    expect(element).toMatchSnapshot();
  });

  it('should apply a .heading class for h1', () => {
    const elementWithH1 = shallow(<Section {...baseProps} level={1} />);

    expect(elementWithH1.find('h1').props().className).toBe('heading');
  });

  it('should apply a .subheading class for h2 and below', () => {
    const elementWithH2 = shallow(<Section {...baseProps} level={2} />);
    const elementWithH3 = shallow(<Section {...baseProps} level={3} />);
    const elementWithH4 = shallow(<Section {...baseProps} level={4} />);
    const elementWithH5 = shallow(<Section {...baseProps} level={5} />);
    const elementWithH6 = shallow(<Section {...baseProps} level={6} />);

    expect(elementWithH2.find('h2').props().className).toBe('subheading');
    expect(elementWithH3.find('h3').props().className).toBe('subheading');
    expect(elementWithH4.find('h4').props().className).toBe('subheading');
    expect(elementWithH5.find('h5').props().className).toBe('subheading');
    expect(elementWithH6.find('h6').props().className).toBe('subheading');
  });

  it('should render a <HTMLContent /> component when raw HTML is passed', () => {
    const props = {
      ...baseProps,
      content: '<p>my dangerous html</p>',
    };
    const element = shallow(<Section {...props} />);

    expect(element.find('HTMLContent')).toHaveLength(1);
    expect(element.find('HTMLContent').props().content).toBe(props.content);
    expect(element.find('p')).toHaveLength(0);
  });

  it('should render the children it receives', () => {
    const props = {
      ...baseProps,
      content: <p>my safe jsx</p>,
    };
    const element = shallow(<Section {...props} />);

    expect(element.find('HTMLContent')).toHaveLength(0);
    expect(element.find('p')).toHaveLength(1);
    expect(element.find('p').props().children).toBe('my safe jsx');
  });
});
