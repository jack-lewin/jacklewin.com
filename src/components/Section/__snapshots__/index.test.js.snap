// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Section
    content="<p>my dangerous html</p>"
    level={3}
    title="Title goes here"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="subheading"
        >
          Title goes here
        </h3>,
        <HTMLContent
          content="<p>my dangerous html</p>"
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Title goes here",
          "className": "subheading",
        },
        "ref": null,
        "rendered": "Title goes here",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": "<p>my dangerous html</p>",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="subheading"
          >
            Title goes here
          </h3>,
          <HTMLContent
            content="<p>my dangerous html</p>"
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Title goes here",
            "className": "subheading",
          },
          "ref": null,
          "rendered": "Title goes here",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": "<p>my dangerous html</p>",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
